app/src/main/java/com/mad41/tripreminder/Login_form.java

```diff
@@ -1,103 +1,318 @@
+                                    readFireBase.start();
+                                } else {
+                                    Toast.makeText(Login_form.this, "Login Failed.", Toast.LENGTH_SHORT).show();
+
+                                }
+
+
+                            }
+                        });
+
+            }
+        });
+
+        Registeration.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                startActivity(new Intent(getApplicationContext(),Registration_Form.class));
+            }
+        });
+
+
+
+    }
+    private void signWithGoogle() {
+        Intent signInIntent = googleSignInClient.getSignInIntent();
+        startActivityForResult(signInIntent,RC_SIGN_IN);
     }
+    private void handleFacebookToken(AccessToken token){
+        Toast.makeText(this, "handleFacebookToken", Toast.LENGTH_SHORT).show();
+        AuthCredential credential = FacebookAuthProvider.getCredential(token.getToken());
+        mAuth.signInWithCredential(credential).addOnCompleteListener(this, new OnCompleteListener<AuthResult>() {
+            @Override
+            public void onComplete(@NonNull Task<AuthResult> task) {
+                if(task.isSuccessful()){
+                    FirebaseUser user = mAuth.getCurrentUser();
+                    Toast.makeText(Login_form.this, "task is Successful", Toast.LENGTH_SHORT).show();
+                    readFireBase.start();
+                   //updateUI(user);
+                    startActivity(new Intent(getApplicationContext(), MainScreen.class));
+                }else{
+                    Toast.makeText(Login_form.this, "failed", Toast.LENGTH_SHORT).show();
+
+                }
+            }
+        });
+    }
+
     @Override
     protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+        mCallbackManager.onActivityResult(requestCode,resultCode,data);
         super.onActivityResult(requestCode, resultCode, data);
-        if(requestCode == AUTH_REQUEST_CODE)
+        if(requestCode == RC_SIGN_IN)
         {
-            IdpResponse response = IdpResponse.fromResultIntent(data);
-            if(requestCode == RESULT_OK ) {
-                Toast.makeText(this, "Done", Toast.LENGTH_SHORT).show();
-                startActivity(new Intent(getApplicationContext(),SecondActivity.class));
-                finish();
-            }
-            else {
-               finish();
-            }
+            Task<GoogleSignInAccount> task = GoogleSignIn.getSignedInAccountFromIntent(data);
+            handleSignInResult(task);
+        }
+    }
+    private void handleSignInResult(Task<GoogleSignInAccount> task) {
+        try {
+            GoogleSignInAccount account = task.getResult(ApiException.class);
+            Toast.makeText(this, "Signed In Successfully", Toast.LENGTH_SHORT).show();
+            FirebaseGoogleAuth(account);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.